src/primitives/point.jl

```diff
@@ -40,17 +40,15 @@ J = Point3f(1, 2, 3) # explicitly ask for single precision
 """
 struct Point{Dim,T} <: Primitive{Dim,T}
   coords::Vec{Dim,T}
-  Point{Dim,T}(coords::Vec{Dim,T}) where {Dim,T} = new{Dim,T}(coords)
+  Point(coords::Vec{Dim,T}) where {Dim,T} = new{Dim,T}(coords)
```

Comment:
Do we need this inner constructor or is it automatically generated by the Julia compiler? Can we try to erase it completely here?

Reply:
The inner constructor is necessary. The constructor is the inference rule for interpreting the implicit `Point` type to the explicit `Point` type. The constructors below cannot replace this.

Reply:
I deleted the line and tested the code, which failed some tests.